### PR TITLE
Introduction to Internet Protocol and Subnetting

### PR DESCRIPTION
**Pull request by kwennB**
#### Prepared introduction to subnetting. 
An in-depth explanation of network terminologies, understanding what a subnet mask is, and a provided tool to calculate it. Learners would understand the differences between the existing internet protocol, routing, and its types. 

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [ ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [ ] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [ ] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [ ] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [ ] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [ ] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [ ] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [ ] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [ ] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by collins-a** [comment]
Hello @kwennB. Welcome to the EngEd program. I'd like to commend you on your article as I can already see it's well written and quite detailed.

However, let's sort out a couple of things before we continue with the review:

1. As a new contributor, I encourage you to add a brief Author Bio (index.md file) and Headshot image (avatar file). Include these files in the authors folder within the GitHub repo via your forked repo similar to the article files so as to be credited for your hard work.

2. Just confirming, is your hero image from Unsplash or any copyright free source? Plus, it is preferred to have an image as close as possible to 600x400 pixels. The current one well exceeds those dimesnions.

Ping me when done.

Again, welcome to EngEd and don't hesitate to check out our [resources](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) pages to make your EngEd experience much easier.

**Comment by kwennB** [comment]
Hello @collins-a.  Thank you for your warm welcome. I'm glad to be a part of this amazing community.

The hero image is from Unsplash. I have just resized the image.
Thank you.

**Comment by collins-a** [comment]
@kwennB Perfect! I'll carry on with the review as soon as I can.

**Comment by kwennB** [comment]
Hello, @collins-a I am to edit the read-marked lines from the article?   Or it has to be updated as it is?
Thank you for your work and time. 

Awaiting your response.

**Comment by collins-a** [comment]
Hi @kwennB. Thanks for your patience, I'd have completed my review sooner but it's been a hectic week. 

At the moment, you don't have to edit anything as I'm yet to do a tech review then when we're done with that we move to the final review.

However, at the moment our EngEd repo has a new structure. As such, articles are getting migrated to a new Articles folder. To make this as easy as possible, and ensure that your article is published as soon as possible:
 
- Download the most recent files off this PR (Make sure to pick the latest one with my revisions and edits).
- Delete your article directory from the articles folders.
- Do a git pull (take a look at the new structure of the repo after a git pull).
- Upload your article files to the [new articles folder](https://github.com/section-engineering-education/engineering-education/tree/master/content/articles).

Please ping me via Slack if you have any problems.

Then we can continue with, and finalize our review.

**Comment by kwennB** [comment]
Thank you for responding. I will get on with it as soon as possible.

I am not on the slack channel, can I get the workspace email or the magic link. 

Thank you.

**Comment by collins-a** [comment]
No worries @kwennB, thanks for raising that, we'll look into the Slack channel issue ASAP cc. @hectorkambow

File: content/articles/introduction-to-subnetting/index.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+| --- | --- | --- | --- | --- | --- | --- | --- |
+        Number(#) of subnets = 22 = 4. 
+        Subnet mask = 2bits = 128 + 64 = 192
+        Range of hosts = 26 = 64   
+
+|        | Range      | Usable Range |
+| ------ | ---------- | ------------ |
+| Network ID | 0-63 | 65 - 126 |
+|     |  64-127 | 129 - 190 |
+| Broadcast Address | 192 - 255 |     |
```
Comment (collins-a) [inline content/articles/introduction-to-subnetting/index.md:135]: The formula is meant to be 2^n + 2, isn't it? Double check the math in this section

File: content/articles/introduction-to-subnetting/index.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+**Routing Protocol**: A routed protocol finds paths (routes) for the IP and the routed protocol. Routing protocols run on Routers, Firewalls, Network servers with Network Operating Systems.
+
+Routing Protocols and the relating Routed Protocol
+| Routing Protocol   | Routed Protocol  |
+| ---------------- | ----------------- |
+| RIP, IGRP, OSPF, EIGRP | IP |
+| RIP, NLSP, EIGRP  | IPX  |
+| RMTP, AURP, EIGRP | Appletalk  |
+
```
Comment (collins-a) [inline content/articles/introduction-to-subnetting/index.md:161]: Consider adding a link to content where a curious reader can read further on these protocols

File: content/articles/introduction-to-subnetting/index.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+Routing is categorized into 3 places; 
+
+i. Static 
+
+ii. Default 
+
+iii. Dynamic
```
Comment (collins-a) [inline content/articles/introduction-to-subnetting/index.md:168]:
```suggestion
Routing is broken down into 3 categories: 

1. Static 

2. Default 

3. Dynamic
```

File: content/articles/introduction-to-subnetting/index.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+#### Advantages and disadvantages of each type of routing
+
+| Type of routing  | Advantages  | Disadvantages  |
+| ---------------- | ----------- | -------------- |
+| Static    | * It has no bandwidth usage between the routers. | * It becomes daunting to handle the chore it is to handle this manually |
+             | * Provides security, only the administrator is allowed to have control over the routing network. | * Lots topography is needed. |
+             | * Has no overhead CPU usage on the router. |
+| Default   | * They establish routes that will never go down | * They require a great deal of CPU power |
+| Dynamic   | * It is easier to configure | * It is less secure compared to static and default routing. |
+            | * It is effective in selecting the best route to a destination network. | It is more expensive in terms of bandwidth and CPU. |
```
Comment (collins-a) [inline content/articles/introduction-to-subnetting/index.md:185]: Well put. However, previewing this table shows that a bit of rearrangement is needed to ensure that it displays correctly.

File: content/articles/introduction-to-subnetting/index.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+            
+**Classless Inter-Domain Routing (CIDR):** CIDR is based off on the idea of subnet masks. It makes it possible to create supernets adding to subnets. CIDR helps extend the number of available addresses. The CIDR stores as a suffix in the IP address like so; 201.105.34.7/24 (/24) is the CIDR in this address.
```
Comment (collins-a) [inline content/articles/introduction-to-subnetting/index.md:187]:
```suggestion
**Classless Inter-Domain Routing (CIDR):** CIDR is based off on the idea of subnet masks. It makes it possible to create supernets adding to subnets. CIDR helps extend the number of available addresses. The CIDR stores as a suffix in the IP address like so; 201.105.34.7/24. (/24) is the CIDR suffix in this address.
```

File: content/articles/introduction-to-subnetting/index.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+### Benefits of Subnetting
+
+- Conservation of IP addresses. 
+- Controls network growth. 
+- Enhances network security. 
+- Easier administration. 
+- Enhances performance and speed.   
+- Reduces network traffic. 
+- It simplifies the network.
+
```
Comment (collins-a) [inline content/articles/introduction-to-subnetting/index.md:200]: I suggest adding a conclusion section to perhaps summarize or let the reader know your final thoughts on the topic you just covered.

**Review by collins-a — COMMENTED** [review]
Hey @kwennB, thank you for your patience with the review. I enjoyed the technical depth and vastness of your article. However, there are a few simple changes to make before approving this for a final review.

I also want you to note how I've represented links within the articles, as it will help you during future submissions.

Brilliant work. Looking forward to future submissions.

**Comment by hectorkambow** [comment]
Thank you v much for your patience as we worked through our Repo re-structuring process. 

The purpose behind this restructure - was to better allow us to handle more volume and allow students to publish more articles and get more involved with the community (with open-source contribution and suggestions). 

With most of the changes coming to completion - we will be sure to get your article published asap. 

Thank you again - and we are looking forward to more amazing articles.

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/introduction-to-subnetting/index.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+- Internet Protocol (IP) Address
+- IPV4 Diagram
+- IP Classes
+- Public vs Private IP
+- Classless Inter-Domain Routing (CIDR)
+- Broadcast ID 
+
+Let’s dive right in.
+
+**Subnetting** is the process of partitioning a network into two or more networks based on utilization. It is a logical process of splitting a single-page network into more strands. Subnetting came into the internet in 1978.
```
[the remark this answers is not in the document]
Comment (kwennB) [reply content/articles/introduction-to-subnetting/index.md:28]: It was 1981. It was certainly a mistake on from my end. 
my article draft on WPS has 1981.

File: content/articles/introduction-to-subnetting/index.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+[Diagram](/introduction-to-subnetting-diagram-of-IPV4.jpeg)
+
+#### IP Classes
+
+Classes are how you can identify network addresses on the internet. The network class an IP address fits into is ascertained by the binary numbers. IP classes are in 5 parts ranging from class A - E. Class A-C are the major used classes.
+
+|class | Range  | Parts of the class | Network IDs | Host IDs | Start address | End address | Leading bits | Size of network number bit field |
+| ---- | ------ | ------------------ | ----------- | -------- | ------------- | ----------- | ------------ | ------------------- |
+@@ -61,19 +71,22 @@ Thirty-two bits (4 x 8) = 4 bytes = 4 octet
+(Robles, 2021)
```
Comment (collins-a) [inline content/articles/introduction-to-subnetting/index.md:63]: It seems like this table is incomplete

File: content/articles/introduction-to-subnetting/index.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+
+To read more on these two protocols, follow this [link](https://www.omnisecu.com/cisco-certified-network-associate-ccna/what-is-the-difference-between-routing-protocols-and-routed-protocols.php).
+
+Routing is then split into 3 categories: 
+
+1. Static 
+2. Default 
+3. Dynamic
+
+**Dynamic Routing(Adaptive):** A router learns dynamic routing by running routed protocols. It adds information to the routing table from connected users immediately.
```
Comment (collins-a) [inline content/articles/introduction-to-subnetting/index.md:136]: Only one type of routing is defined here

File: content/articles/introduction-to-subnetting/index.md [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+
+A particular address consists of; 
+
+1. Host identity 
+
+2. Network identity.
+
+Below is a diagram of IPV4
+
+[Diagram](/introduction-to-subnetting-diagram-of-IPV4.jpeg)
```
Comment (collins-a) [inline content/articles/introduction-to-subnetting/index.md:54]: For the article to look as presentable as possible, is it possible to get a clearer image?

**Review by collins-a — COMMENTED** [review]
Hey @kwennB, thank you for making the required changes. Good job! However, just a couple of points to sort and we are ready for the final review. Also, always remember to carry out a Grammarly check after making changes.

Ping me when done.

**Review by collins-a — APPROVED** [review]
Hey @kwennB, well done on a very informative first article.

A couple of things to note for your future writing:
1. Voice and tone. Be sure to involve the reader more where necessary. Make it seem like you're communicating with the reader. It not only humanizes the content but also shows more relatability and makes content more approachable. Furthermore, communicating your own thoughts (where applicable), especially in the conclusion section of the article strengthens your point of view of the subject.
2. Formatting, in form of representing links and numbering of points. This can be noted in some of the changes I made.

I encourage you to explore our [resources](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) page to help with the two points above. Looking forward to future articles!

**Comment by ahmadmardeni1** [comment]
Hey @kwennB 👋
I'm doing the final review for your article, please take a look at the pictures below, the tables are not rendered well on our website, please provide pictures instead of the tables.

Capture

**Comment by hectorkambow** [comment]
@kwennB great work - please go through the final edits I made to see where future articles can be improved. 

cc @collins-a

**Comment by kwennB** [comment]
> @kwennB great work - please go through the final edits I made to see where future articles can be improved.
> 
> cc @collins-a

@hectorkambow  Thank you.

**Pull request merged**